### DISCUSSION TITLE
No module named mnamer.__main__; 'mnamer'

### DISCUSSION THREAD

**Discussion by SweetthD3ViL** [jkwill87/mnamer#137] [Questions]
Hello there,

After updating my ARM Archlinux NAS yesterday, I am having issue with mnamer.

- terminal output:

> $ python -m mnamer -bvr /mnt/hd/Download/
> /usr/bin/python: No module named mnamer.__main__; 'mnamer' is a package and cannot be directly executed

I did installed mnamer originally using "sudo pip3 install mnamer" and wanted to uninstalling it to reinstall it using your advised method of pipx. But now "sudo pip3 uninstall mnamer" returns "WARNING: Skipping mnamer as it is not installed."

Doing a bit of research it seems to be in relation to python 3.9?

So, I would prefer to be able to a) uninstall mnamer globally, and then re-install it using pipx. And then how do we run mnamer with the new python 3.9?

Thank you

**Comment by jkwill87** [comment]
> Doing a bit of research it seems to be in relation to python 3.9?

I suspect that its probably not Python 3.9 that's the problem exactly, its that you likely have multiple versions of Python, Pip, and/or site-packages for multiple versions of Python.

Not knowing all the details I would maybe try the following to clean things up:

- run `python -m site` and ensure that an mnamer directory isn't in any of the directories listed
- run `which mnamer` and delete the directory containing mnamer if it exists
- uninstall pipx if you installed it using pip and install it using `sudo pacman install python-pipx` to ensure its added to your PATH correctly
- run `pipx install mnamer`
- run mnamer just using the `mnamer` command

**Comment by SweetthD3ViL** [comment]
Hi jkwill87,

Thank you for your reply, I search all trace of mnamer on my system and deleted it all to clean it. pipx is installed using pacman. 
I then installed mnamer using "pipx install mnamer" 

Now see below terminal result when trying to execute it. 

```
sweetth@goflex ~]$ python -m mnamer -bvr /mnt/hd/Download/
/usr/bin/python: No module named mnamer
[sweetth@goflex ~]$ mnamer -bvr /mnt/hd/Download/
-bash: mnamer: command not found
```

**Comment by SweetthD3ViL** [comment]
Hi jkwill87,

Sorry for above query, I should have spend the time to read about pipx. All sorted now and working well. Thank you!
